File: trunk/superstreetfire/src/gamemodel/player.py
'''
player.py

Holds the class that represents a player in the Super Street Fire Game.
Players have a number (1 or 2) and a total number of hitpoints (which is also
a percentage of their total life).

@author: Callum Hay
'''
import logging

class Player:
    LOGGER_NAME    = 'player_logger'
    MAX_HIT_POINTS = 100
    
    def __init__(self, playerNum):
        assert(playerNum == 1 or playerNum == 2)
        self.playerNum    = playerNum
        self._logger      = logging.getLogger(Player.LOGGER_NAME)
        self.Reset()
        
    def Reset(self):
        self.ResetHealth()
        self.numRoundWins = 0
        self.isInvincible = False
    
    def ResetHealth(self):
        self.hitPoints = Player.MAX_HIT_POINTS
    
    def IsKnockedOut(self):
        return (self.hitPoints <= 0)
    
    def DoDamage(self, hpDamage):
        if self.isInvincible:
            return
        self.hitPoints -= hpDamage
        self._logger.debug("Player " + str(self.playerNum) + " has been damaged (damage amt = " + \
                           str(hpDamage) + ", life left = " + str(self.hitPoints) + ")")
        
def GetOtherPlayerNum(playerNum):
    if playerNum == 1:
        return 2
    elif playerNum == 2:
        return 1
    else:
        assert(False)

File: trunk/ssf-java/GameModel/src/main/java/ca/site3/ssf/gamemodel/Player.java
package ca.site3.ssf.gamemodel;

/**
 * The Player class represents a player of the Super Street Fire game.
 * @author Callum
 *
 */
public class Player {
	
	//public enum PlayerNumber { PLAYER_ONE, PLAYER_TWO }
	
	public static final float KO_HEALTH   = 0.0f;
	public static final float FULL_HEALTH = 100.0f;
	
	private float health;
	private int numRoundWins;
	private boolean isInvincible;
	
	public Player() {
		this.reset();
		this.isInvincible = false;
	}
	
	public void reset() {
		this.resetHealth();
		this.numRoundWins = 0;
	}
	
	public void resetHealth() {
		this.health = Player.FULL_HEALTH;
	}
	
	public void doDamage(float damageAmt) {
		assert(damageAmt > 0);
		if (this.isInvincible) {
			return;
		}
		
		this.health -= damageAmt;
	}
	
	public void setInvincible(boolean invincibilityOn) {
		this.isInvincible = invincibilityOn;
	}
	
	public boolean isKOed() {
		return (this.health <= Player.KO_HEALTH);
	}
	
	public float getHealth() {
		return this.health;
	}
	
	public int getNumRoundWins() {
		return this.numRoundWins;
	}
	
	
	
	
}
